pymbolic/primitives.py

```diff
@@ -683,6 +734,124 @@ def __iter__(self):
         raise TypeError("expression types are not iterable")
 
 
+# {{{ dataclasses support
+
+def augment_expression_dataclass(cls: type[DataclassInstance]) -> type[Expression]:
```

Comment:
Could this also be fancy and add the `mapper_method` based on the class name? https://stackoverflow.com/a/1176023

I think that may interfere with a fallback in some mapper? So maybe not possible..

Reply:
I've run with this, and it certainly makes the expression node classes look nice and clean. I *think* it's also safe, in that it allows the class to override the default, while encouraging deletion when the explicit assignment matches the default.

Reply:
It turns out that how this interacts with inheritance is a bit janky. It will automatically give a `mapper_method` to every class, and then other classes may inherit that. Please take a look at the logic that I've got now; I'm not totally sure how I should feel about it.

Reply:
Hm, the logic makes sense to me.. at least I can't think of any example where that would be undesired.

Could those checks be simplified to `"mapper_method" in cls.__dict__`? Not sure what the difference between that and `hasattr` is wrt to inheritance :\

Reply:
Yep, that's a better check all around. Thanks! I've changed it to that and added a comment to explain the inheritance semantics, so that that knowledge isn't lost. (see next push)

Reply:
Looks good all around now! Thanks! :rocket: